lib/cybersource/api.rb

```diff
@@ -276,6 +279,20 @@ def should_credit?(kb_payment_id, context, options = {})
         (now - transaction.created_at) >= threshold
       end
 
+      def before_gateways(kb_transaction, last_transaction, payment_source, amount_in_cents, currency, options, context = nil)
+        # Provide necessary information for merchant descriptor
+        if options[:merchant_descriptor].present? && options[:merchant_descriptor]['is_amex'].nil?
```

Comment:
> This is the most difficult part of this PR, for our use case, the credit card information will only passed through the authorize call (actually the combo call), and no card information will be in the capture, refund and void call.

Understood, but I don't think the user of the API should have to bother filling-in these details. Let's chat offline about it.